pkg/api/authz.go

```diff
@@ -282,7 +282,7 @@ func AuthzHandler(ctlr *Controller) mux.MiddlewareFunc {
 			// for extensions we only need to know if the user is admin and what repos he can read, so run next()
 			if request.RequestURI == fmt.Sprintf("%s%s", constants.RoutePrefix, constants.ExtCatalogPrefix) ||
 				strings.Contains(request.RequestURI, constants.FullSearchPrefix) ||
-				isMgmtRequested {
+				isMgmtRequested || strings.Contains(request.RequestURI, constants.FullPreferencesPrefix) {
```

Comment:
authZ/authN? for extensions? who has read-only access? what happens when anonymous authN is enabled?

Reply:
Can we write unit tests for access control scenarios?

Reply:
We have these tests here: pkg/extensions/search/user_preferences_test.go at line 140

Reply:
For Unauthenticated users the tests start at line 346. 

I will tag you in a comment to that line, I can't link it here